### ISSUE TITLE
Talend SDI (7.1) in Talend 7.3 - no components in palette

### ISSUE THREAD

**Issue by lucvanlinden** [talend-spatial/talend-spatial#80]
Similar to ticket 79, in 7.3 following install instructions, (copy SDI plugin into plugins folder) does not show up the SDI plugins in the palette.

Is this (7.3) not supported (yet)?
Is (7.1) not working with 7.3??
Is it a constraints to have the version number match?

tx

Luc

**Comment by mtravis** [comment]
I'm seeing the same issue.

**Comment by archaeogeek** [comment]
I believe that you need to match versions of Talend and Talend-Spatial. 7.1 works fine (but I experienced the same problem trying to use Talend-Spatial 7.1 in Talend 7.3)

**Comment by fxprunayre** [comment]
> I believe that you need to match versions of Talend and Talend-Spatial. 7.1 works fine (but I experienced the same problem trying to use Talend-Spatial 7.1 in Talend 7.3)

Not really, we can consider that the Talend-Spatial version number is the minimum version required on Talend side. And at some point, the module does not work with latest Talend version once a changed occur on the plugin loading mechanism.

I confirm that 7.3.1 does not look to load any of the Talend Spatial plugins (no workspace/XYZ/.spatial folder, no generic schema > create from Shapefile or OGR, no Geo group in palette, no routines in code which are basically the 4 plugins to loads). No errors in `workspace/.metadata/.log` so not that easy to track. Maybe @mhirt would have suggestions on a change on the studio that we have to apply to spatial plugins?

**Comment by lucvanlinden** [comment]
Maybe these should become "custom components", although must admit I do not
have any idea what that means technically, nor about impact?

**Comment by fxprunayre** [comment]
> Maybe these should become "custom components", although must admit I do not

Talend spatial is not only about components, it adds some menu for schema, it adds a Geometry type in Talend internals, Routines.

After quick hack in 7.3.1, it looks like adding `,org.talend.libraries.sdi-7.1.1@4,org.talend.sdi.designer.components-7.1.1@4,org.talend.sdi.designer.routines-7.1.1@4,org.talend.sdi.repository.ui.actions.metadata-7.1.1@4,org.talend.sdi.repository.ui.actions.metadata.ogr-7.1.1@4,org.talend.sdi.workspace.spatial-7.1.1@4` in `configuration/config.ini` for `osgi.bundles=...` the components are loaded.

Then there are exception in routines regarding Geotools dependency (I think we solved or workaround this  @archaeogeek ?).

Some testing welcomed so we can document a way of making it work. I'll ask Talend team if they can explain how to improve that.

**Comment by anacaona83** [comment]
Hello, I do have a similar problem with an 7.2 TOS DI installation and SDI 7.1. I did add config.ini osgi.bundles but still don't find spatial components. My SDI files are located in C:\talend\TOS_DI-Win32-20190620_1446-V7.2.1\plugins
I don't know if I have to to something more (for instance in "parameters"/talend/component or else). Here ascreenshot of my plugins. Thanks a lot!
![TOS_SDI_installation](https://user-images.githubusercontent.com/12633071/108853193-b7b5b980-75e6-11eb-89ca-2a17e8804872.gif)
